src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1682,13 +1584,42 @@ public unsafe void HandleEvents(Interop.Sys.SocketEvents events)
 
             if ((events & Interop.Sys.SocketEvents.Read) != 0)
             {
-                _receiveQueue.Complete(this);
+                _receiveQueue.HandleEvent(this);
             }
 
             if ((events & Interop.Sys.SocketEvents.Write) != 0)
             {
-                _sendQueue.Complete(this);
+                _sendQueue.HandleEvent(this);
             }
         }
+
+        //
+        // Tracing stuff
+        //
+
+        // To enabled tracing:
+        // (1) Add reference to System.Console in the csproj
+        // (2) #define SOCKETASYNCCONTEXT_TRACE
+
+#if SOCKETASYNCCONTEXT_TRACE
+        public const bool TraceEnabled = true;
+#else
+        public const bool TraceEnabled = false;
```

Comment:
Yeah, this would be better.  At some point we may want to turn these into actual ETW events, which could be enabled in release builds too.  But unless/until we do that, I think Conditional is better.